components/places/android/src/main/java/org/mozilla/places/PlacesConnection.kt

```diff
@@ -67,18 +71,26 @@ class PlacesConnection(path: String, encryption_key: String? = null) : PlacesAPI
     }
 
     override fun getVisited(urls: List<String>): List<Boolean> {
-        val urlsToJson = JSONArray()
-        for (url in urls) {
-            urlsToJson.put(url)
-        }
-        val urlStr = urlsToJson.toString()
-        val visitedStr = rustCallForString { error ->
-            LibPlacesFFI.INSTANCE.places_get_visited(this.db!!, urlStr, error)
+        val urlStrings = StringArray(urls.toTypedArray(), "utf8")
+        val byteBuffer = ByteBuffer.allocateDirect(urls.size)
```

Comment:
Here you're using direct buffer because of JNA limitations, right?

Reply:
Also, might be worth mentioning in a comment that since JNA's StringArray extends `Memory`, its `size` is actually its size in bytes, not number of elements in the array as it would be normally.

Reply:
... and isn't your use of `size` wrong? `allocateDirect` takes a capacity in bytes, and you're giving it number size of the url array in bytes, which would be much, much larger than necessary.

Reply:
No, I think this is right. You'd be right if i had done `urlStrings.size`.

Reply:
This is quite a potential footgun though

Reply:
Ahhh, you're right. Yeah, perhaps some renaming for clarity is in order! Similar names totally threw me off.

Reply:
And I'm using a direct buffer because JNA requires it, yes. Doing it like this also allows us to avoid an extra copy.